sdk/core/src/main/java/com/klaviyo/core/KLog.kt

```diff
@@ -0,0 +1,116 @@
+package com.klaviyo.core
+
+import android.os.Build
+import com.klaviyo.core.config.Config
+import com.klaviyo.core.config.Log
+import com.klaviyo.core.config.Log.Level
+import java.util.regex.Pattern
+
+open class KLog : Log {
+
+    private var _logLevel: Level? = null
+    override var logLevel: Level
+        get() {
+            if (_logLevel == null && Registry.isRegistered<Config>()) {
```

Comment:
Why check isRegistered here?

Reply:
This is checking if `applicationContext` will be available through `Config`. Without it, we can't read from the manifest.

Reply:
Otherwise, accessing `Registry.Config` would throw an exception, then we'd try to log the exception... ♾️